jujugui/static/gui/src/app/components/generic-button/_generic-button.scss

```diff
@@ -1,102 +1,198 @@
+        background-color: transparent;
+      }
     }
-}
+  }
 
-.button-row {
-    .generic-button {
-        width: 100%;
-        padding-left: 5px;
-        padding-right: 5px;
-    }
+  .button--inline-neutral {
+    display: inline-block;
+    width: auto;
+  }
+}
 
-    &--multiple .generic-button {
-        margin-right: 10px;
+/// Positive button styling
+/// @group Buttons
+/// @example
+/// <a class="button--positive">
+///   ...
+/// </a>
+@mixin gui-positive-button {
+  .button--positive,
+  .button--inline-positive {
+    @extend %button-pattern;
+    color: $white;
+    background-color: $success;
 
-        &:last-child {
-            margin-right: 0;
-        }
+    &:hover {
+      background-color: darken($success, 10%);
     }
 
-    &--count-2 .generic-button {
-        width: calc(50% - 5px);
+    &:disabled,
+    &.button--disabled {
+      &:hover {
+        background-color: $success;
+      }
     }
+  }
 
-    &--count-3 .generic-button {
-        width: calc(33.333% - 7px);
-    }
+  .button--inline-positive {
+    display: inline-block;
+    width: auto;
+  }
 }
 
-.deployment-bar-panel .generic-button {
-    &--type-confirm {
-        border-top-left-radius: 0;
-        border-bottom-left-radius: 0;
+
+/// Destructive button styling
+/// @group Buttons
+/// @example
+/// <a class="button--destructive">
+///   ...
+/// </a>
+@mixin gui-destructive-button {
+  .button--destructive,
+  .button--inline-destructive {
+    @extend %button-pattern;
+    color: $white;
+    background-color: $error;
+
+    &:hover {
+      background-color: darken($error, 10%);
     }
 
-    &--type-blue {
-        border-top-right-radius: 0;
-        border-bottom-right-radius: 0;
-        margin-right: 1px;
+    &:disabled,
+    &.button--disabled {
+      &:hover {
+        background-color: $error;
+      }
     }
+  }
+
+  .button--inline-destructive {
+    display: inline-block;
+    width: auto;
+  }
 }
 
-.deployment-summary .generic-button {
-    padding-left: 60px;
-    padding-right: 60px;
 
-    &--type-clear-changes {
-        float: left;
+/// Base button styling
+/// @group Buttons
+/// @example
+/// <a class="button--base">
+///   ...
+/// </a>
+@mixin gui-base-button {
+  .button--base,
+  .button--inline-base {
+    @extend %button-pattern;
+    color: $cool-grey;
+    background-color: transparent;
+
+    &:hover {
+      background-color: $light-grey;
     }
-}
 
-.entity-details .generic-button {
-    width: 100%;
-}
+    &:disabled,
+    &.button--disabled {
+      &:hover {
+        background-color: transparent;
+      }
+    }
+  }
 
-.inspector-config .generic-button {
-    padding: 10px 10px;
+  .button--inline-base {
+    display: inline-block;
+    width: auto;
+  }
 }
 
-.inspector-current-version__item {
-    .generic-button {
-        display: none;
-        float: right;
-        margin-top: -3px;
-        padding: 5px 10px;
-    }
 
-    &:focus,
-    &:active,
+/// Deployment button styling
```

Comment:
I'm not familiar with the triple slash comment style in scss - any difference/advantage over `/*`?